src/ChainSafe.GamingSdk.Gelato/GelatoModule.cs

```diff
@@ -0,0 +1,247 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using ChainSafe.GamingSDK.EVM.Web3.Core;
+using ChainSafe.GamingSDK.EVM.Web3.Core.Evm;
+using ChainSafe.GamingSdk.Gelato.Dto;
+using ChainSafe.GamingSdk.Gelato.Types;
+using ChainSafe.GamingWeb3;
+using ChainSafe.GamingWeb3.Environment;
+using Nethereum.ABI.EIP712;
+using Nethereum.Hex.HexTypes;
+using Web3Unity.Scripts.Library.Ethers.Providers;
+using Web3Unity.Scripts.Library.Ethers.Signers;
+
+namespace ChainSafe.GamingSdk.Gelato
+{
+    public class GelatoModule : IGelatoModule, ILifecycleParticipant
+    {
+        private readonly GelatoClient gelatoClient;
+        private readonly IRpcProvider provider;
+        private readonly GelatoConfig config;
+        private readonly IChainConfig chainConfig;
+
+        public GelatoModule(IHttpClient httpClient, IChainConfig chainConfig, GelatoConfig config, IRpcProvider provider)
+        {
+            this.gelatoClient = new GelatoClient(httpClient, config);
+            this.provider = provider;
+            this.config = config;
+            this.chainConfig = chainConfig;
+        }
+
+        public async Task<RelayResponse> CallWithSyncFee(CallWithSyncFeeRequest request)
+        {
+            try
+            {
+                request.ChainId = int.Parse(chainConfig.ChainId);
+                return await this.gelatoClient.Post<CallWithSyncFeeRequest, RelayResponse>(RelayCall.CallWithSyncFee, request);
+            }
+            catch (System.Exception e)
+            {
+                throw new Exception($"GelatoRelaySDK/relayWithSyncFee: Failed with error: ${e.Message}");
+            }
+        }
+
+        public async Task<RelayResponse> CallWithSyncFeeErc2771(CallWithSyncFeeErc2771Request request, ISigner wallet)
```

Comment:
You can remove this overload and change `RelayRequestOptions options` parameter to `RelayRequestOptions options = null` in the next one

Reply:
Also, why not define ISigner as a dependency in the constructor?

Reply:
@oleksandrchainsafe because its only needed for the 2 functions which request it and not the functions that don't request the signer

Reply:
If we want to support the case when a user doesn't register ISigner, but still wants to use GelatoModule we could create two constructors for GelatoModule. One expects the ISigner dependency and the other one that doesn't. You could then check if the ISigner field is not null inside the functions, that require it and throw exceptions in this case.
Still, I don't think anybody would need to use GelatoModule without ISigner being bound as this means that there are no active user/account/keys, therefore you can't send transactions.

Reply:
There are a number of use cases where someone would like to trigger a transaction without having a distinct user signing the function in mind. 

I will repeat, Gelato is a transaction relaying infrastructure which allows the developer to fund transactions of the users. 

This means, if you're using Gelato, **The user is never sending or signing any transactions**